### PR TITLE
Seacas: Update for latest seacas releaes version

### PR DESCRIPTION
**Pull request by gsjaardema**
Add option to pull latest SEACAS release; update dependency version information.

### CODE REVIEW

File: var/spack/repos/builtin/packages/seacas/package.py [hunk 1]
```diff
@@ -133,6 +136,7 @@ class Seacas(CMakePackage):
 
     # ###################### Dependencies ##########################
     depends_on("cmake@3.17:", type="build")
+    depends_on("cmake@3.22:", when="@2023-10-24", type="build")
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/seacas/package.py]:
```suggestion
    depends_on("cmake@3.22:", when="@2023-10-24:", type="build")
    depends_on("cmake@3.17:", when="@:2023-05-30", type="build")
```
These should probably be ranges, and dependencies are usually ordered so newer versions are before older ones.

File: var/spack/repos/builtin/packages/seacas/package.py [hunk 2]
```diff
@@ -143,6 +147,7 @@ class Seacas(CMakePackage):
     depends_on("fmt@8.1.0:9", when="@2022-03-04:2022-05-16")
     depends_on("fmt@9.1.0", when="@2022-10-14")
     depends_on("fmt@9.1.0:", when="@2023-05-30")
+    depends_on("fmt@10.1.0:", when="@2023-10-24")
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/seacas/package.py]: Same comments here,

File: var/spack/repos/builtin/packages/seacas/package.py (hunk 1 above)
Comment (gsjaardema) [reply var/spack/repos/builtin/packages/seacas/package.py]: done i think.  Is `master` considered after `2023-10-24`?

File: var/spack/repos/builtin/packages/seacas/package.py (hunk 2 above)
Comment (gsjaardema) [reply var/spack/repos/builtin/packages/seacas/package.py]: Done i think.

**Review by sethrj — APPROVED** [review]
Lgtm

**Pull request merged**